### PR TITLE
feat: support the failure/success threshold for probe

### LINKED ISSUE #230: failure/success threshold before sending notifications
**Background & X Problem**
It's easy to have false postitives when looking at CPUs/latencies. For example, CPU might have a peak at 80%+ but just for a few seconds.

**Proposal and Expectation**
Just like failure-threshold in https://github.com/TwiN/gatus, it's handy if I'm allowed to set the threshold before sending notifications.

### PR DESCRIPTION
**Pull request by haoel**
This PR supports the failure/success threshold features - we can configure the number of continuously failed/successful probes to determine whether the endpoint status is DOWN or UP.

for example:

```yaml
probe:
    failure: 2 # number of consecutive failed probes needed to determine the status down, default: 1
    success: 1 # number of consecutive successful probes needed to determine the status down, default: 1
```

close #230

### CODE REVIEW

File: docs/Manual.md [hunk 1]
```diff
@@ -627,7 +627,8 @@ The following example configurations illustrate the EaseProbe supported features
 
 - `timeout` - the maximum time to wait for the probe to complete. default: `30s`.
 - `interval` - the interval time to run the probe. default: `1m`.
-
+- `failure` - number of continuously failed probes needed to determine the status down, default: 1
+- `success` - number of continuously successful probes needed to determine the status down, default: 1
```
Comment (localvar) [inline docs/Manual.md:631]:
```suggestion
- `success` - number of continuously successful probes needed to determine the status up, default: 1
```

Comment (localvar) [inline docs/Manual.md:631]:
```suggestion
- `success` - number of continuously successful probes needed to determine the status up, default: 1
```
there are other similar cases.

File: global/probe.go [hunk 2]
```diff
@@ -19,10 +19,19 @@ package global
 
 import "time"
 
+// StatusChangeThresholdSettings is the settings for probe threshold
+type StatusChangeThresholdSettings struct {
+	// the failures threshold such as 2, 5
+	Failure int `yaml:"failure,omitempty" json:"failure,omitempty" jsonschema:"description=the failures threshold to change the status such as 3, 5,default=1"`
```
Comment (localvar) [inline global/probe.go:25]: comma in the description may result in wrong-generated documentation.

Comment (haoel) [reply global/probe.go]: very good finding!

File: docs/Manual.md [hunk 3]
```diff
@@ -627,7 +627,8 @@ The following example configurations illustrate the EaseProbe supported features
 
 - `timeout` - the maximum time to wait for the probe to complete. default: `30s`.
 - `interval` - the interval time to run the probe. default: `1m`.
-
+- `failure` - number of continuously failed probes needed to determine the status down, default: 1
+- `success` - number of continuously successful probes needed to determine the status up, default: 1
```
Comment (proditis) [inline docs/Manual.md:631]:
```suggestion
- `failure` - number of consecutive failed probes needed to determine the status down, default: 1
- `success` - number of consecutive successful probes needed to determine the status up, default: 1
```

File: docs/Manual.md [hunk 4]
```diff
@@ -1308,6 +1309,9 @@ settings:
   probe:
     timeout: 30s # the time out for all probes
     interval: 1m # probe every minute for all probes
+    failure: 2 # number of continuously failed probes needed to determine the status down, default: 1
+    success: 1 # number of continuously successful probes needed to determine the status up, default: 1
+
```
Comment (proditis) [inline docs/Manual.md:1314]:
```suggestion
    failure: 2 # number of consecutive failed probes needed to determine the status down, default: 1
    success: 1 # number of consecutive successful probes needed to determine the status up, default: 1

```

File: resources/config.yaml [hunk 5]
```diff
@@ -42,6 +42,8 @@
 #       expression: "x_time('//feed/updated') > '2022-07-01'" # the expression to evaluate.
 #     # configuration
 #     timeout: 10s # default is 30 seconds
+#     failure: 2 # number of continuously failed probes needed to determine the status down, default: 1
+#     success: 1 # number of continuously successful probes needed to determine the status up , default: 1
```
Comment (proditis) [inline resources/config.yaml:46]:
```suggestion
#     failure: 2 # number of consecutive failed probes needed to determine the status down, default: 1
#     success: 1 # number of consecutive successful probes needed to determine the status up , default: 1
```

File: resources/config.yaml [hunk 6]
```diff
@@ -57,6 +59,8 @@ http: # http probes
 #     interval: 2m # default is 60 seconds
 #     proxy: socks5://proxy.server:1080 # Optional. Only support socks5.
 #                                       # Also support the `ALL_PROXY` environment.
+#     failure: 2 # number of continuously failed probes needed to determine the status down, default: 1
+#     success: 1 # number of continuously successful probes needed to determine the status up, default: 1
```
Comment (proditis) [inline resources/config.yaml:63]:
```suggestion
#     failure: 2 # number of consecutive failed probes needed to determine the status down, default: 1
#     success: 1 # number of consecutive successful probes needed to determine the status up, default: 1
```

File: resources/config.yaml [hunk 7]
```diff
@@ -386,6 +390,8 @@ notify:
 #   probe:
 #     timeout: 30s # the time out for all probes
 #     interval: 1m # probe every minute for all probes
+#     failure: 2 # number of continuously failed probes needed to determine the status down, default: 1
+#     success: 1 # number of continuously successful probes needed to determine the status up, default: 1
```
Comment (proditis) [inline resources/config.yaml:394]:
```suggestion
#     failure: 2 # number of consecutive failed probes needed to determine the status down, default: 1
#     success: 1 # number of consecutive successful probes needed to determine the status up, default: 1
```

**Review by proditis — COMMENTED** [review]
Minor change of continuously to consecutive. 
Need to test run some cases still.

File: docs/Manual.md (hunk 3 above)
Comment (haoel) [inline docs/Manual.md:631]:
```suggestion
- `failure` - number of consecutive failed probes needed to determine the status down, default: 1
- `success` - number of consecutive successful probes needed to determine the status up, default: 1
```

**Review by zhao-kun — APPROVED** [review]

**Review by samanhappy — APPROVED** [review]

**Pull request merged**